sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -302,16 +307,32 @@ public BlobInputStream openInputStream(BlobInputStreamOptions options) {
         options = options == null ? new BlobInputStreamOptions() : options;
         ConsistentReadControl consistentReadControl = options.getConsistentReadControl() == null
             ? ConsistentReadControl.ETAG : options.getConsistentReadControl();
-
-        BlobProperties properties = getPropertiesWithResponse(options.getRequestConditions(), null, null).getValue();
-        String eTag = properties.getETag();
-        String versionId = properties.getVersionId();
+        BlobRequestConditions requestConditions = options.getRequestConditions() == null
+            ? new BlobRequestConditions() : options.getRequestConditions();
 
         BlobRange range = options.getRange() == null ? new BlobRange(0) : options.getRange();
         int chunkSize = options.getBlockSize() == null ? 4 * Constants.MB : options.getBlockSize();
 
-        BlobRequestConditions requestConditions = options.getRequestConditions() == null
-            ? new BlobRequestConditions() : options.getRequestConditions();
+        com.azure.storage.common.ParallelTransferOptions pOptions =
+            new com.azure.storage.common.ParallelTransferOptions().setBlockSizeLong((long) chunkSize);
+        BiFunction<BlobRange, BlobRequestConditions, Mono<BlobDownloadAsyncResponse>> downloadFunc =
+            (r, conditions) -> client.downloadWithResponse(r, null, conditions, false);
+        Tuple3<Long, BlobRequestConditions, BlobDownloadAsyncResponse> tuple =
+            ChunkedDownloadUtils.downloadFirstChunk(range, pOptions, requestConditions, downloadFunc, true).block();
+        if (tuple == null) {
+            throw logger.logExceptionAsError(new IllegalStateException("Downloading first chunk returned null"));
+        }
```

Comment:
how can this happen?

Reply:
It can't because we always emit a value, but ci interprets a call to block as possibly returning null and requires a check

Reply:
which tool reports that? is it possible to suppress this warning inline @alzimmermsft ?

Reply:
`.block()` in `Mono` and `Flux` is annotated as `nullable` so Spotbugs will flags any variables instantiated using `.block()` results in it being flagged as potential null value usage.

Reply:
Would be great to avoid it altogether. However, it would be great to know the pattern to suppress warnings if we have good reason to do so (like writing dead code in this code). 
In .NET we use [this](https://github.com/Azure/azure-sdk-for-net/search?q=%22%23pragma+warning+disable%22) to accomplish that. I guess something similar should be possible with java's toolchain?

Reply:
Updated